source/Scrapers/TeamCityBuildScraper.cs

```diff
@@ -32,27 +32,27 @@ protected override void Scrape()
             teamCityClient.ConnectWithAccessToken(teamCityToken);
 
             var hungBuilds = teamCityClient.Builds
-                .GetFields("count,build(id,probablyHanging,buildTypeId)")
-                .ByBuildLocator(BuildLocator.WithDimensions(running: true), new List<string> { "hanging:true" })
+                .GetFields("count,build(id,probablyHanging,buildTypeId,composite)")
+                .ByBuildLocator(BuildLocator.WithDimensions(running: true), new List<string> { "hanging:true", "composite:false" })
                 .ToArray();
 
-            var gauge = metricFactory.CreateGauge("probably_hanging_builds", "Count of running builds that appear to be hung", "buildTypeId");
+            var gauge = metricFactory.CreateGauge("probably_hanging_builds", "Running builds that appear to be hung", labelNames: new []{"buildTypeId", "buildId"});
 
-            foreach (var build in hungBuilds.GroupBy(x => x.BuildTypeId))
+            foreach (var build in hungBuilds)
             {
-                gauge.WithLabels(build.Key).Set(build.Count());
-                Logger.Debug("Build Type {BuildTypeId}, Count {Count}", build.Key, build.Count());
+                gauge.WithLabels(new []{build.BuildTypeId, build.Id}).Set(1);
+                Logger.Debug("Build Type {BuildTypeId}, build ID {BuildId} has hung", build.BuildTypeId, build.Id);
             }
 
-            var currentBuildTypes = hungBuilds.Select(x => x.BuildTypeId).Distinct().ToArray();
-            seenBuildTypes.UnionWith(currentBuildTypes);
-            var absentBuildTypes = seenBuildTypes.Except(currentBuildTypes);
+            var currentBuilds = hungBuilds.Select(x => (x.BuildTypeId, x.Id)).ToArray();
+            seenBuilds.UnionWith(currentBuilds);
+            var absentBuildTypes = seenBuilds.Except(currentBuilds);
 
-            foreach (var item in absentBuildTypes)
+            foreach (var (buildType, buildId) in absentBuildTypes)
```

Comment:
Nitpick for consistency
```suggestion
            foreach (var (buildType, buildId) in absentBuilds)
```

Reply:
Done!